src/api/cpp/cvc5.h

```diff
@@ -1167,11 +1167,30 @@ class CVC5_EXPORT Term
    *       for example, the term f(x, y) will have Kind APPLY_UF and three
    *       children: f, x, and y
    */
-  class const_iterator : public std::iterator<std::input_iterator_tag, Term>
+  class const_iterator
   {
     friend class Term;
 
    public:
+    /* The following types are required by trait std::iterator_traits */
+
+    /** Iterator tag */
+    using iterator_category = std::input_iterator_tag;
```

Comment:
Disclaimer: I'm not an expert on this whole iterators business.
After reading https://en.cppreference.com/w/cpp/iterator/iterator_tags (and related), I think we should rather use `forward_iterator_tag`.
This is a general thing: input iterator means that the thing that is pointed to is (or may be) invalidated after you increment the iterator (I guess the analogy would be reading from `stdin` or so). We can however do things like `auto it = it2 = ...; ++it; auto x = *it2;`.
This comment most probably holds for all iterators in this PR.

Reply:
Hmm... I will think about it. actually I'm not familiar with this either...